pkg/controller/gardenletdeployer/actuator.go

```diff
@@ -429,64 +432,103 @@ func (a *Actuator) checkSeedSpec(ctx context.Context, spec *gardencorev1beta1.Se
 }
 
 func (a *Actuator) reconcileSeedSecrets(ctx context.Context, obj client.Object, spec *gardencorev1beta1.SeedSpec, gardenletDeployment *seedmanagementv1alpha1.GardenletDeployment) error {
-	// If backup is specified, create or update the backup secret if it doesn't exist or is owned by the object
+	if spec.Backup == nil {
+		return nil
+	}
+
+	// If backup is specified and DoNotCopyBackupCredentials feature gate is disabled,
+	// create or update the backup secret if it doesn't exist or is owned by the object.
 	// TODO(vpnachev): Add support for WorkloadIdentity
-	if spec.Backup != nil {
-		var checksum string
-
-		// Get backup secret
-		backupSecret, originalErr := kubernetesutils.GetSecretByObjectReference(ctx, a.GardenClient, spec.Backup.CredentialsRef)
-		if originalErr == nil {
-			checksum = utils.ComputeSecretChecksum(backupSecret.Data)[:8]
-		} else if client.IgnoreNotFound(originalErr) != nil {
-			return originalErr
+	var (
+		checksum     string
+		allowCopying = !utilfeature.DefaultFeatureGate.Enabled(features.DoNotCopyBackupCredentials)
+	)
+
+	// Get backup secret
+	backupSecret, originalErr := kubernetesutils.GetSecretByObjectReference(ctx, a.GardenClient, spec.Backup.CredentialsRef)
+
+	if client.IgnoreNotFound(originalErr) != nil {
+		return originalErr
+	}
+
+	if apierrors.IsNotFound(originalErr) && !allowCopying {
+		return fmt.Errorf("the configured backup secret does not exist, however the feature gate DoNotCopyBackupCredentials is enabled and shoot infrastructure credentials will not be reused: %w", originalErr)
+	}
+
+	if originalErr == nil {
+		checksum = utils.ComputeSecretChecksum(backupSecret.Data)[:8]
+	}
+
+	const secretStatusLabel = "secret.backup.gardener.cloud/status"
+
+	// Create or update backup secret if it doesn't exist, or is owned by the object, or was previously managed by this controller
+	if allowCopying && (apierrors.IsNotFound(originalErr) || metav1.IsControlledBy(backupSecret, obj) || backupSecret.Labels[secretStatusLabel] == "previously-managed") {
```

Comment:
Does it make sense to extract the label value as const next to the label key?
I would even extract them to `pkg/apis/core/v1beta1/constants/types_constants.go` but I am also fine if you keep them inline here.